src/fdc/MSXtar.cc

```diff
@@ -31,11 +31,19 @@ namespace openmsx {
 
 static constexpr unsigned FREE_FAT = 0x000;
 static constexpr unsigned FIRST_CLUSTER = 0x002;
-static constexpr unsigned BAD_FAT = 0xFF7;
-static constexpr unsigned EOF_FAT = 0xFFF; // actually 0xFF8-0xFFF, signals EOF in FAT12
+static constexpr unsigned MAX_FAT12 = 0xFF6;
+static constexpr unsigned BAD_FAT12 = 0xFF7;
+static constexpr unsigned EOF_FAT12 = 0xFFF; // actually 0xFF8-0xFFF, signals EOF in FAT12
+static constexpr unsigned MAX_FAT16 = 0xFFF6;
+static constexpr unsigned BAD_FAT16 = 0xFFF7;
+static constexpr unsigned EOF_FAT16 = 0xFFFF; // actually 0xFFF8-0xFFFF, signals EOF in FAT16
+static constexpr unsigned EOF_FAT = 0x0FFFFFFF; // coinciding with FAT32's EOF
```

Comment:
Just checking: you use `EOF_FAT` as a stand-in to either mean `EOF_FAT12` or `EOF_FAT16`, right?
So technically any numerical value > 0xffff would do, and you picked the value 0x0fffffff (which happens to be the EOF value for FAT32, but that doesn't really matter here).
This is fine, I just want to make sure I understand.

Reply:
Yep.